cmd/livepeer_cli/wizard_transcoder.go

```diff
@@ -185,3 +194,68 @@ func (w *wizard) callReward() {
 	fmt.Printf("Calling reward for round %v\n", c)
 	httpGet(fmt.Sprintf("http://%v:%v/reward", w.host, w.httpPort))
 }
+
+func (w *wizard) vote() {
+	if w.offchain {
+		glog.Error("Can not vote in 'offchain' mode")
+		return
+	}
+
+	fmt.Print("Enter the contract address for the poll you want to vote on")
+	poll := w.readStringAndValidate(func(in string) (string, error) {
+		if !ethcommon.IsHexAddress(in) {
+			return "", fmt.Errorf("invalid hex address address=%v", in)
+		}
+		return in, nil
+	})
+
+	var (
+		confirm = "n"
+		choice  = 99
+	)
+
+	for confirm == "n" {
+		choice = 99
+		opts := w.showVoteOptions()
+		if opts == nil {
+			return
+		}
+
+		for _, ok := voteOptions[choice]; !ok; {
+			fmt.Printf("Enter the ID of the option you want to vote for")
+			choice = w.readInt()
+			if _, ok := voteOptions[choice]; ok {
+				break
+			}
+			fmt.Println("Must enter a valid ID")
+		}
+
+		fmt.Printf("Are you sure you want to vote \"%v\" ? (y/n)", voteOptions[choice])
+		confirm = w.readStringYesOrNo()
+	}
+
+	data := url.Values{
+		"poll":     {poll},
+		"choiceID": {fmt.Sprintf("%v", choice)},
+	}
+
+	result := httpPostWithParams(fmt.Sprintf("http://%v:%v/vote", w.host, w.httpPort), data)
+
+	fmt.Printf("\n %v \n", result)
+}
+
+func (w *wizard) showVoteOptions() map[int]string {
+	opts := make([]string, 3)
+	for i, opt := range voteOptions {
+		opts[i] = opt
+	}
```

Comment:
This is kind of hacky but it ensures ordering

Reply:
An alternative could be to use a slice to list the options and represent the option as an enum type:

```golang
const voteOptions = []VoteOption{VoteOption.Yes, ...}

func (v VoteOption) String() string {
    switch v {
        case VoteOption.Yes:
            return "Yes"
        ...
    }
}
```

An invalid vote option would return an empty string for the `String()` call - could event wrap this into `ValidVoteOption()` function for improved readability.